src/rest.js

```diff
@@ -0,0 +1,34 @@
+import axios from 'axios';
+import cookies from 'js-cookie';
+
+export default class RestClient {
+  constructor(apiRoot = '/api/v1', { useCookie = true }) {
+    this._axios = axios.create();
+    this._axios.defaults.baseURL = apiRoot;
+
+    if (useCookie) {
+      this.token = cookies.get('girderToken');
+    }
+  }
+
+  /**
+   * Use this to make Web API requests.
+   * @returns An Object with the same API as axios.
+   */
+  get request() {
+    return this._axios;
+  }
+
+  get token() {
+    return this._token;
```

Comment:
wouldn't  it be better to just return `this._axios.defaults.headers.common['Girder-Token'] = val` to prevent storing the data in two places?